File: ios/Classes/banner/AdropBannerViewFactory.swift
import Foundation
import Flutter
import UIKit

class AdropBannerViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    private var bannerManager: AdropBannerManager
    
    init(messenger: FlutterBinaryMessenger, bannerManager: AdropBannerManager) {
        self.messenger = messenger
        self.bannerManager = bannerManager
        super.init()
    }
    
    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        let call = CallCreateAd(encoding: args as? [String : Any?])
        if let banner = bannerManager.getAd(unitId: call.unitId) {
            return AdropFlutterPlatformView(view: banner)
        } else {
            return ErrorView()
        }
    }
    
    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

private class ErrorView: NSObject, FlutterPlatformView {
    
    func view() -> UIView {
        return UIView()
    }
    
}

File: android/src/main/kotlin/io/adrop/adrop_ads/banner/AdropBannerViewFactory.kt
package io.adrop.adrop_ads.banner

import android.content.Context
import android.view.View
import io.adrop.adrop_ads.model.CallCreateAd
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class AdropBannerViewFactory(
    private val viewManager: AdropBannerManager
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val callData = CallCreateAd(args as? Map<String, Any?>)

        val banner = viewManager.getAd(callData.unitId, callData.requestId)
        return if (banner == null) {
            return ErrorView(context)
        } else {
            return FlutterPlatformView(banner)
        }
    }
}

private class ErrorView(val context: Context) : PlatformView {

    override fun getView(): View {
        return View(context)
    }

    override fun dispose() {}
}
